w3ctrace/state.go

```diff
@@ -19,13 +19,34 @@ type State []string
 func ParseState(traceStateValue string) (State, error) {
 	var state State
 
-	for _, st := range strings.SplitN(traceStateValue, ",", 32) {
-		st = strings.TrimSpace(st)
-		if st == "" {
-			continue
+	// comasLimit is used to prevent handling of malformed values that contain a lot of comas
+	const comasLimit = 256
+	rawEntries := strings.SplitN(traceStateValue, ",", comasLimit)
+
+	rawEntriesLen := len(rawEntries)
+	if rawEntriesLen > 32 {
+		filtered := 0
+		for _, st := range rawEntries {
+			if rawEntriesLen-filtered <= 32 || len(state) == 32 {
+				break
+			}
+
+			st = strings.TrimSpace(st)
```

Comment:
This would potentially modify list members now owned by Instana, which may break trace correlation downstream. I'd suggest we don't alter the value that we did not add to the state.

Reply:
I tried to keep the same behavior as before https://github.com/instana/go-sensor/blob/master/w3ctrace/state.go#L23.

Reply:
This does not look correct either 😅

Reply:
:) Let's change it then.

Reply:
Yes, but according to the spec the `value` may include whitespace characters (`%x20`) and since they are indistinguishable from optional whitespace (`OWS`) in the end, I'd suggest we don't modify the elements that we don't add.